src/transform.php

```diff
@@ -28,12 +28,18 @@ function ($carry, $fn) {
  * the result.
  *
  * @param string $methodName
- * @param array  $arguments
+ * @param array  $args
  *
  * @return callable
  */
 function callMethod($methodName, ...$args)
 {
+    if (!is_string($methodName)) {
```

Comment:
Agreed. Maybe using a project specific extension of `InvalidArgumentException` might be a good idea?